sdc/str_arr_ext.py

```diff
@@ -1438,3 +1433,13 @@ def append_string_array_to(result, pos, A):
         j += 1
 
     return i
+
+
+@numba.njit(no_cpython_wrapper=True)
+def make_str_arr_from_list(str_list):
+
+    data_total_chars = np.sum(np.asarray([len(s) for s in str_list]))
```

Comment:
Why not avoid converting list to the array and just get sum from the list of lengths? Something like that:
`data_total_chars = sum(len(s) for s in str_list)`

Reply:
@densmirn I tried that, but it doesn't work in nopython mode (see [Numba python builtin functions](https://numba.pydata.org/numba-doc/dev/reference/pysupported.html#built-in-functions). 
Also the line
`data_total_chars = sum(len(s) for s in str_list)`
requires the use of generators in a closure call, which is not supported either.

But you're probably right and there's no need for conversion here, so I changed it to explicit prange.